shared/reducers/fs.js

```diff
@@ -116,42 +116,32 @@ const updatePathItem = (
   }
 }
 
-const haveSamePartialSyncConfig = (tlf1: Types.Tlf, tlf2: Types.Tlf) =>
-  tlf2.syncConfig &&
-  tlf1.syncConfig &&
-  tlf2.syncConfig.mode === 'partial' &&
-  tlf1.syncConfig.mode === 'partial' &&
-  tlf2.syncConfig.enabledPaths.equals(tlf1.syncConfig.enabledPaths)
-
-const updateTlf = (oldTlf?: ?Types.Tlf, newTlf: Types.Tlf): Types.Tlf => {
-  if (!oldTlf) {
-    return newTlf
-  }
-  // TODO: Ideally this should come in with other data from the same RPC.
-  const newTlfDontClearSyncConfig = newTlf.syncConfig ? newTlf : newTlf.set('syncConfig', oldTlf.syncConfig)
-  if (
-    !I.is(newTlfDontClearSyncConfig.syncConfig, oldTlf.syncConfig) &&
-    !haveSamePartialSyncConfig(oldTlf, newTlfDontClearSyncConfig)
-  ) {
-    return newTlfDontClearSyncConfig
-  }
-  if (!newTlf.resetParticipants.equals(oldTlf.resetParticipants)) {
-    return newTlfDontClearSyncConfig
-  }
-  if (!newTlf.conflict.equals(oldTlf.conflict)) {
-    return newTlfDontClearSyncConfig
-  }
-  // syncConfig, resetParticipants, and conflict all stayed thte same in value,
-  // so just reuse old reference.
-  return oldTlf.merge(
-    newTlfDontClearSyncConfig.withMutations(n =>
-      n
-        .set('syncConfig', oldTlf.syncConfig)
-        .set('resetParticipants', oldTlf.resetParticipants)
-        .set('conflict', oldTlf.conflict)
-    )
-  )
-}
+const updateTlf = (oldTlf?: ?Types.Tlf, newTlf: Types.Tlf): Types.Tlf =>
+  oldTlf
+    ? oldTlf.merge(
+        newTlf.withMutations(n =>
+          n
+            .set(
+              'syncConfig',
+              // This comes from a different RPC for now. So make sure we don't
+              // override non-nil syncConfig with a nil one.
+              !newTlf.syncConfig || I.is(newTlf.syncConfig, oldTlf.syncConfig)
```

Comment:
Can you put parentheses around this line to make order of operations clearer? I assumed at first that the `||` was being an "if this is not null then that" thing.

Reply:
I think my prettier/linter remove them usually, but I’ll try and see if I am make it clearer